File: Thawbuk-store/lib/domain/entities/category.dart
import 'package:equatable/equatable.dart';

class Category extends Equatable {
  final String id;
  final String name;
  final String? description;
  final String image;
  final String createdBy;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Category({
    required this.id,
    required this.name,
    this.description,
    required this.image,
    required this.createdBy,
    required this.createdAt,
    required this.updatedAt,
  });

  @override
  List<Object?> get props => [
        id,
        name,
        description,
        image,
        createdBy,
        createdAt,
        updatedAt,
      ];
}

File: src/domain/entity/category.ts
import { CategoryInfoDTO } from "../../application/dtos/category.dto";

export interface ICategory {
  _id: string;
  name: string;
  nameAr: string;
  description: string | null;
  descriptionAr: string | null;
  image: string;
  productsCount?: number;
  createdBy: string,
  createdAt?: Date;
  updatedAt?: Date;
}

export class CategoryMapper {
  static toDTO(category: ICategory): CategoryInfoDTO {
    return {
      _id: category._id,
      name: category.name,
      nameAr: category.nameAr,
      description: category.description,
      descriptionAr: category.descriptionAr,
      image: category.image,
      productsCount: category.productsCount,
    };
  }
}
